### ISSUE TITLE
Implement hyperparameter search techniques in utils

### ISSUE THREAD

**Issue by ljvmiranda921** [ljvmiranda921/pyswarms#4]

**Comment by SioKCronin** [comment]
I'd like to help with this. Is it ok if we add sklearn as a dependency to take advantage of their GridSearchCV and RandomizedSearchCV?

**Comment by ljvmiranda921** [comment]
Hi, sorry I haven't done any commits lately, currently on a trip. I am also thinking about that. But if we are to use scikit-learn's hyperparameter search tools, maybe we can just scrap this feature altogether? Unless we provide a more convenient API in using them then it is fine. What do you think?

**Comment by SioKCronin** [comment]
I compared scikit-learn's implementation of GridSearch and RandomizedSearch with Optunity's, and I think it should be straightforward to implement these here. I haven't found another PSO library that has this functionality, and I imagine people would be more apt to tune if it was built in. I can start with GridSearch and see how it goes?

**Comment by ljvmiranda921** [comment]
I see I see. Sure! Let's try GridSearch first and iterate from there. 

Thanks a lot! @SioKCronin!

**Comment by SioKCronin** [comment]
Great. I should be able to get this going next week, and will reach out if I hit any snags.
